aws/cloudformation-templates/template.yaml

```diff
@@ -174,6 +177,12 @@ Parameters:
       is set to 'Yes'.
     Default: "AWS Retail Demo Store"
   
+  PinpointSMSLongCode:
+    Type: String
+    Description: >
+      Phone number(a dedicated long code) for Amazon Pinpoint to send and receive messages at. Be sure to enable two way SMS for this long code using Pinpoint. This parameter is required when 'Auto-Configure Pinpoint'
+      is set to 'Yes'.
```

Comment:
@james-jory and @shivani  : Do you think it would make sense to also have an option to include/exclude SMS integration for Pinpoint?

Reply:
I think if we make it optional, it is up to the customers to opt in for SMS integration right? @james-jory I set the default value of the long code to NONE. We are anyways checking that the long code is not NONE in the auto workshop before creating the segment, campaign and template.

Reply:
> @james-jory and @ShivaniMehendarge : Do you think it would make sense to also have an option to include/exclude SMS integration for Pinpoint?

I think as long as the app functionality, pre-create lambda, and workshop code has defensive logic when a long code is not available/NONE, we should be fine. For example, we know that with Event Engine all CFN params must have values and that separate long codes can't be pre-provisioned for all accounts ahead of time. So if the event deployer chooses the Pinpoint auto-create, long codes won't be available and SMS functionality should be appropriately disabled. But a missing/NONE long code value should not cause any fatal issues either. However, if later I set a long code in SSM, either manually or in the workshop and redeploy the webapp, the SMS functionality should automatically become active. This is how several other parts of the app work today. For example, the app works fine without Personalize campaigns but once they're created, immediately starts using them. Same with Search and Lex. Make sense?

Reply:
> @alexchirayath I think if we make it optional, it is up to the customers to opt in for SMS integration right? @james-jory I set the default value of the long code to NONE. We are anyways checking that the long code is not NONE in the auto workshop before creating the segment, campaign and template.

This sounds good. As long as the web app, backend, pre-create lambda, and workshops work appropriately with and without a long code, we are in good shape.

Reply:
Yup